File: data/src/main/java/com/smb/data/authentication/SocialNetworkType.java
package com.smb.data.authentication;

/**
 * Created by dev on 25.01.18.
 */

public enum SocialNetworkType implements SocialNetworkCodes {
    FACEBOOK {
        @Override
        public int code() {
            return 1;
        }
    }, TWITTER {
        @Override
        public int code() {
            return 2;
        }
    }, GOOGLE {
        @Override
        public int code() {
            return 3;
        }
    }, INSTAGRAM {
        @Override
        public int code() {
            return 4;
        }
    }
}

File: data/src/main/java/com/smb/data/authentication/SocialNetworkType.kt
package com.smb.data.authentication

import com.smb.core.authentication.SocialNetworkCodes

/**
 * Created by dev on 25.01.18.
 */

enum class SocialNetworkType : SocialNetworkCodes {
    FACEBOOK {
        override fun code(): Int {
            return 1
        }
    },
    TWITTER {
        override fun code(): Int {
            return 2
        }
    },
    GOOGLE {
        override fun code(): Int {
            return 3
        }
    },
    INSTAGRAM {
        override fun code(): Int {
            return 4
        }
    }
}
